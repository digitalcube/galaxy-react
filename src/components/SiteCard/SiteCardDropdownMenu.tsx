import React, { FC, ReactNode, useState, PropsWithChildren } from 'react';
import { FaEllipsisH } from 'react-icons/fa';
import { Dropdown, DropdownMenu, DropdownToggle } from 'reactstrap';

export type SiteCardDropdownMenuProps = PropsWithChildren<{
  icon?: ReactNode;
  right?: boolean;
}>;
export const SiteCardDropdownMenu: FC<SiteCardDropdownMenuProps> = ({
  icon,
  children,
  right = true,
}) => {
  const [open, setOpen] = useState(false);
  const toggle = () => setOpen(!open);
  return (
    <Dropdown
      isOpen={open}
      toggle={toggle}
      className="ml-4 site-control"
      direction="down"
    >
      <DropdownToggle
        className="dropdown-toggle"
        role="button"
        data-toggle="dropdown"
        aria-haspopup={true}
        aria-expanded={false}
        tag="span"
      >
        <i aria-hidden="true">{icon || <FaEllipsisH className="d-block" />}</i>
      </DropdownToggle>
      <DropdownMenu right={right}>{children}</DropdownMenu>
    </Dropdown>
  );
};
