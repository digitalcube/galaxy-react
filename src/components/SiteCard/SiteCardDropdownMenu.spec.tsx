import { render } from '@testing-library/react';
import React from 'react';
import { Default as SiteCardDropdownMenu } from './SiteCardDropdownMenu.stories';

describe('SiteCardDropdownMenu', () => {
  it('to match snapshot [column %p | item %p]', () => {
    const { asFragment } = render(
      <SiteCardDropdownMenu {...(SiteCardDropdownMenu.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
