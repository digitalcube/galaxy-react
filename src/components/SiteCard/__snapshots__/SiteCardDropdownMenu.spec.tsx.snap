// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteCardDropdownMenu to match snapshot [column %p | item %p] 1`] = `
<DocumentFragment>
  <div
    class="sites-list"
  >
    <section
      class="d-flex flex-column site-list-body"
    >
      <div
        class="ml-4 site-control dropdown"
      >
        <span
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle"
          data-toggle="dropdown"
          role="button"
        >
          <i
            aria-hidden="true"
          >
            <svg
              class="d-block"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
              />
            </svg>
          </i>
        </span>
        <div
          aria-hidden="true"
          class="dropdown-menu dropdown-menu-right"
          role="menu"
          tabindex="-1"
        >
          <button
            class="dropdown-item px-3 py-2 d-flex align-items-center justify-content-between dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Start WordPress
            <i
              aria-hidden="true"
            >
              <svg
                class="d-block"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
                />
              </svg>
            </i>
          </button>
          <button
            class="dropdown-item px-3 py-2 d-flex align-items-center justify-content-between dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Restart WordPress
            <i
              aria-hidden="true"
            >
              <svg
                class="d-block"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                />
              </svg>
            </i>
          </button>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
